src/Serialization/main.jl

```diff
@@ -86,22 +86,27 @@ for (T, str) in (
     AbstractAlgebra.Generic.MPolyRing{Hecke.NfRelNSElem{nf_elem}} => "AbstractAlgebra.Generic.MPolyRing{Hecke.NfRelNSElem{nf_elem}}",
     AbstractAlgebra.Generic.MPolyRing{nf_elem} => "AbstractAlgebra.Generic.MPolyRing{nf_elem}",
     AbstractAlgebra.Generic.MPolyRing{NfAbsNSElem} => "AbstractAlgebra.Generic.MPolyRing{NfAbsNSElem}",
+    AbstractAlgebra.Generic.MPolyRing{padic} => "AbstractAlgebra.Generic.MPolyRing{padic}",
+    AbstractAlgebra.Generic.MPoly{padic} => "AbstractAlgebra.Generic.MPoly{padic}",
     AbstractAlgebra.Generic.Poly{AbstractAlgebra.Generic.Frac{fmpq_poly}} => "AbstractAlgebra.Generic.Poly{AbstractAlgebra.Generic.Frac{fmpq_poly}}",
     AbstractAlgebra.Generic.Poly{Hecke.NfRelElem{nf_elem}} => "AbstractAlgebra.Generic.Poly{Hecke.NfRelElem{nf_elem}}",
     AbstractAlgebra.Generic.Poly{Hecke.NfRelNSElem{nf_elem}} => "AbstractAlgebra.Generic.Poly{Hecke.NfRelNSElem{nf_elem}}",
     AbstractAlgebra.Generic.Poly{nf_elem} => "AbstractAlgebra.Generic.Poly{nf_elem}",
     AbstractAlgebra.Generic.Poly{NfAbsNSElem} => "AbstractAlgebra.Generic.Poly{NfAbsNSElem}",
+    AbstractAlgebra.Generic.Poly{padic} => "AbstractAlgebra.Generic.Poly{padic}",
     AbstractAlgebra.Generic.PolyRing{AbstractAlgebra.Generic.Frac{fmpq_poly}} => "AbstractAlgebra.Generic.PolyRing{AbstractAlgebra.Generic.Frac{fmpq_poly}}",
     AbstractAlgebra.Generic.PolyRing{Hecke.NfRelElem{nf_elem}} => "AbstractAlgebra.Generic.PolyRing{Hecke.NfRelElem{nf_elem}}",
     AbstractAlgebra.Generic.PolyRing{Hecke.NfRelNSElem{nf_elem}} => "AbstractAlgebra.Generic.PolyRing{Hecke.NfRelNSElem{nf_elem}}",
     AbstractAlgebra.Generic.PolyRing{nf_elem} => "AbstractAlgebra.Generic.PolyRing{nf_elem}",
     AbstractAlgebra.Generic.PolyRing{NfAbsNSElem} => "AbstractAlgebra.Generic.PolyRing{NfAbsNSElem}",
+    AbstractAlgebra.Generic.PolyRing{padic} => "AbstractAlgebra.Generic.PolyRing{padic}",
```

Comment:
Generic serialization for `AbstractAlgebra.Generic.(M)Poly(Ring)` would be nice, and also for `MPolyIdeal`. Not for this PR, of course! But do you plan to work on that?

Reply:
planning to start tackling these this week